mrmap/monitoring/models.py

```diff
@@ -164,12 +178,20 @@ class Meta:
         verbose_name = _('Monitoring result')
         verbose_name_plural = _('Monitoring results')
 
+    @property
+    def resource(self):
+        return self._resource.get_target(self)
+
+    @resource.setter
+    def resource(self, value):
+        self._resource.set_target(self, value)
+
     @property
     def icon(self):
         return Tag(tag='i', attrs={"class": [IconEnum.MONITORING_RESULTS.value]}).render()
 
     def get_absolute_url(self):
-        return reverse('monitoring:result_details', args=[self.uuid, ])
+        return reverse('monitoring:result_details', args=[self.id, ])
```

Comment:
inherit from GenericModelMixin and fix the url namings to get url building with our common solution.